### PR TITLE
Reviving adapter to a working state with Fluent version 21.2.0

### PR DESCRIPTION
**Pull request by IshaanDesai**
This PR consists of a wide range of modifications to the legacy version of the adapter to revive it and get it to work with Fluent version 21.2.0. A working state of the adapter was achieved by @mtree22 recently.

The goal of this PR is to polish the adapter code and documentation and push a working and well-documented state to the develop branch, from which eventually a new adapter can be released.

@mtree22 lets use this PR to work on the tasks that we discussed.

### CODE REVIEW

File: examples/Cavity2D/steer-fluent.txt [hunk 1]
```diff
@@ -0,0 +1,10 @@
+; commands to send to Fluent to initialize and solve FSI analysis
```
Comment (MakisH) [inline examples/Cavity2D/steer-fluent.txt:1]: A `README.md` file here would be very helpful.

**Comment by vierousa** [comment]
examples/Cavity2D 。 I'm not familiar with Python, so I would like to know how to use CSMdummy.py. Thank you very much

**Comment by IshaanDesai** [comment]
Hi @vierousa,

The file `CSMdummy.py` is a computational structural mechanics dummy solver. It reads forces from preCICE, computes displacements, and writes them back to preCICE. As it is a dummy solver, displacements are capped at `0.002`. You run this script by directly calling it: `python3 CSMdummy.py`. The example itself is a [lid-driven cavity flow with a flexible bottom](https://precice.discourse.group/t/lid-driven-cavity-with-flexible-bottom/364).

The Cavity2D example does not have documentation of its own, but all the necessary files are arranged in a way that you can refer to the [How to start Fluent](https://github.com/precice/fluent-adapter/blob/fluent_21_2_0/README.md#22-how-to-start-fluent) section of the repository README, and run the case using the instructions in there. I know this is not ideal, but this is work in progress still.

Please let us know if you face further issues in running the Cavity2D example.

**Comment by IshaanDesai** [comment]
There has been significant interest in the adapter recently, and the branch of this PR is the closest working state, and also fairly documented. I propose merging this pull request and continuing development on the develop branch, mainly to prevent the use of a feature branch as a develop branch.

**Review by MakisH — APPROVED** [review]
As long as this helps further development, I am happy to see it merged! :rocket: 

I have no idea about any licensing issues, please confirm before merging.

**Review by BenjaminRodenberg — APPROVED** [review]
Also sounds like a good approach to me. Let's merge this one as it has de-facto become the develop branch anyways :smirk:

**Pull request merged**